.github/workflows/androidTestSuite.yml

```diff
@@ -0,0 +1,133 @@
+name: Android Test Suite
+
+# The workflow will start for every PR against master branch
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+# Multiple jobs run in parallel
+jobs:
+  android_test_cases:
+    name: Android Unit Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkouts the current branch for processing
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Setup the Java environment
+      - name: Setup JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+
+      # Cache and restore the Gradle dependencies on multiple runs
+      - name: Gradle cache
+        uses: gradle/gradle-build-action@v2
+
+      # Run android unit tests & organize test reports
+      - name: Run tests
+        run: |
+          chmod +x gradlew
+          ./gradlew jacocoTestProdDebugUnitTestReport
+          ./gradlew copyUnitTestBuildArtifacts
+
+      # Uploads the folder or file in path to Github
+      - name: Upload Test Report
+        # Run this step even if the previous fails
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Unit Test Report
+          path: artifacts
+
+      # Upload the test report to Codecov for coverage report
+      - name: Codecov Report
```

Comment:
Codecov needs to install in the test environment, so install before the execution of the unit test cases. better to set up before  `Setup JDK`

Reply:
I have used the [codecov action](https://github.com/marketplace/actions/codecov) instead of a local one to tackle this possible problem.